### PR TITLE
Eagle 1229

### PR DESCRIPTION
**Pull request by M-Wicenec**
## Summary by Sourcery

This pull request refactors the relationship checks between nodes in Edge.ts, introduces a new utility method to disable loop awareness on edges, updates the edge stroke type handling for loop-aware edges, and removes an obsolete method from Node.ts.

- **Enhancements**:
    - Refactored relationship checks between source and destination nodes in Edge.ts to improve clarity and maintainability.
    - Introduced a new method fixDisableEdgeLoopAware in Utils.ts to disable loop awareness on edges.
    - Updated edgeGetStrokeType method in GraphRenderer.ts to handle loop-aware edges with a new stroke type.
- **Chores**:
    - Removed the hasLocalPortWithId method from Node.ts as it is no longer needed.

### CODE REVIEW

File: src/Edge.ts [hunk 1]
```diff
@@ -393,43 +394,26 @@ export class Edge {
...
-                parentIsEFN = eagle.logicalGraph().findNodeByKey(sourceNode.getParentKey()).getCategory() === Category.ExclusiveForceNode;
-            }
-        }
-
-        // if a node is connecting to its parent, it must connect to the local port
-        if (isParent && !destinationNode.hasLocalPortWithId(destinationPortId)){
-            Edge.isValidLog(edgeId, Eagle.LinkValid.Invalid, Errors.Show("Source port is connecting to its parent, yet destination port is not local", function(){Utils.showEdge(eagle, edgeId);}), showNotification, showConsole, errorsWarnings);
-        }
+        //these checks are to see if the source or destination node are embedded apps whose parent is a sibling of the other source or destination node
+        const destPortIsEmbeddedAppOfSibling : boolean =  sourceNode.getParentKey() !== null && destinationNode.getEmbedKey() != null && sourceNode.getParentKey() === eagle.logicalGraph().findNodeByKeyQuiet(destinationNode.getEmbedKey()).getParentKey()
```
[the remark this answers is not in the document]
Comment (M-Wicenec) [reply src/Edge.ts]: fixed using ?.

File: src/Edge.ts [hunk 2]
```diff
@@ -393,43 +394,26 @@ export class Edge {
...
+        const srcPortIsEmbeddedAppOfSibling : boolean =  destinationNode.getParentKey() !== null && sourceNode.getEmbedKey() != null && destinationNode.getParentKey() === eagle.logicalGraph().findNodeByKeyQuiet(sourceNode.getEmbedKey()).getParentKey()
```
[the remark this answers is not in the document]
Comment (M-Wicenec) [reply src/Edge.ts]: you already said this above

File: src/Utils.ts [hunk 3]
```diff
@@ -1753,6 +1753,10 @@ export class Utils {
         eagle.logicalGraph().removeEdgeById(edgeId);
     }
 
+    static fixDisableEdgeLoopAware(eagle: Eagle, edgeId: string): void {
+        eagle.logicalGraph().findEdgeById(edgeId).setLoopAware(false)
```
[the remark this answers is not in the document]
Comment (M-Wicenec) [reply src/Utils.ts:1757]: again fixed using ?.

**Review by james-strauss-uwa — APPROVED** [review]
Looks great, just a few minor code formatting issues

File: src/Edge.ts [hunk 4]
```diff
@@ -440,9 +424,21 @@ export class Edge {
             }
         }
 
+        //checking if the edge is un-neccessarily loopAware
```
Comment (james-strauss-uwa) [inline src/Edge.ts:427]: There is just one 'c' in necessarily

**Pull request merged**